starter/src/main/java/com/vivy/springdoc/OpenAPIAutoConfiguration.java

```diff
@@ -32,4 +50,66 @@ public OpenAPI bearerOpenAPI() {
+
+        return new ApplicationRunner() {
+            @Override
+            public void run(ApplicationArguments args) throws Exception {
+                int apiDocPort = environment.getProperty("local.server.port", Integer.class, -1);
+                String openApiPath = getEnvVariable("openApiEndpoint.path", String.class, "/v3/api-docs");
+                UriBuilder uriBuilder = uriFactory.builder().scheme("http").host("localhost").path(openApiPath).port(apiDocPort);
+                ResponseEntity<String> apiDocsResponse = getRestTemplate().getForEntity(uriBuilder.build(), String.class);
+                if (apiDocsResponse.getStatusCode() == HttpStatus.OK) {
+                    String apiDocFileName = getEnvVariable("file.name", "api-docs.json");
+                    String apiDocFilePath = getEnvVariable("file.path", "build");
+                    String responseBody = apiDocsResponse.getBody();
+                    responseBody = replaceLocalEndpoint(responseBody, uriBuilder.replacePath("").build());
+                    File apiDocFile = createApiDocFile(apiDocFileName, apiDocFilePath);
+                    writeDocumentationInFile(apiDocFile, responseBody);
+                }
+            }
+
+            private String replaceLocalEndpoint(String body, URI localUri) {
+                String protocol = getEnvVariable("url.protocol", "http");
+                String host = getEnvVariable("url.host", "localhost");
+                int port = getEnvVariable("url.port", Integer.class, -1);
+                UriBuilder externalBaseUrlBuilder = uriFactory.builder().scheme(protocol).host(host).port(port);
+                body = body.replaceAll(localUri.toString(), externalBaseUrlBuilder.build().toString());
+                return body;
+            }
+
+            private File createApiDocFile(String apiDocFileName, String apiDocFilePath) throws IOException {
+                Path root = FileSystems.getDefault().getPath("").toAbsolutePath();
+                Path filePath = Paths.get(root.toString(), apiDocFilePath, apiDocFileName);
+                Files.createDirectories(filePath.getParent());
+                //noinspection ResultOfMethodCallIgnored, result is irrelevant
+                filePath.toFile().createNewFile();
+                return filePath.toFile();
+            }
+
+            private String getEnvVariable(String envVariable, String defaultValue) {
+                return environment.getProperty(APPLICATION_PROPERTIES_PREFIX + envVariable, defaultValue);
+            }
+
+            private <T> T getEnvVariable(String envVariable, Class<T> targetType, T defaultValue) {
+                return environment.getProperty(APPLICATION_PROPERTIES_PREFIX + envVariable, targetType, defaultValue);
+            }
+
+            private void writeDocumentationInFile(File apiDocFile, String body) throws IOException {
+                FileOutputStream outputStream = new FileOutputStream(apiDocFile);
```

Comment:
looks like you don't close the file - that could lead to file descriptor leaks, please use Files.write static methods, that would be safer

Reply:
done, by a intelij suggestion i have change the closing of the file to `Automatic Resource Management in Java`, pls have one more look after the code is commited if this closes the file, as I haven't used it before. Link https://www.geeksforgeeks.org/automatic-resource-management-java/